plugins/modules/route53.py

```diff
@@ -607,6 +604,9 @@ def main():
         'HealthCheckId': health_check_in,
         'SetIdentifier': identifier_in,
     })
+    if command_in == 'delete' and aws_record is not None:
+        resource_record_set['TTL'] = aws_record.get('TTL')
+        resource_record_set['ResourceRecords'] = aws_record.get('ResourceRecords')
```

Comment:
```suggestion
        if not resource_record_set['ResourceRecords']:
            resource_record_set['ResourceRecords'] = aws_record.get('ResourceRecords')
```

I'd only overwrite ResourceRecords if they've not explicitly passed entries, to support the partial deletion of round-robin results.